tapfreighter/chain_porter.go

```diff
@@ -148,37 +150,103 @@ func (p *ChainPorter) Start() error {
 
 		// Start the main chain porter goroutine.
 		p.Wg.Add(1)
-		go p.assetsPorter()
+		go p.mainEventLoop()
 
-		// Identify any pending parcels that need to be resumed and add
-		// them to the exportReqs channel so they can be processed by
-		// the main porter goroutine.
-		ctx, cancel := p.WithCtxQuit()
-		defer cancel()
-		outboundParcels, err := p.cfg.ExportLog.PendingParcels(ctx)
-		if err != nil {
-			startErr = err
-			return
-		}
-
-		// We resume delivery using the normal parcel delivery mechanism
-		// by converting the outbound parcels into pending parcels.
-		for idx := range outboundParcels {
-			outboundParcel := outboundParcels[idx]
-			log.Infof("Attempting to resume delivery for "+
-				"anchor_txid=%v",
-				outboundParcel.AnchorTx.TxHash().String())
-
-			// At this point the asset porter should be running.
-			// It should therefore pick up the pending parcels from
-			// the channel and attempt to deliver them.
-			p.exportReqs <- NewPendingParcel(outboundParcel)
-		}
+		startErr = p.resumePendingParcels()
 	})
 
 	return startErr
 }
 
+// resumePendingParcels attempts to resume delivery for any pending parcels that
+// were previously interrupted. This is done by querying the export log for any
+// pending parcels and adding them to the outboundParcels channel so they can be
+// processed by the main porter goroutine.
+func (p *ChainPorter) resumePendingParcels() error {
```

Comment:
Non-blocking note: should this logic be in its own goroutine?

IIUC, right now this blocks `Start()` from returning. If there are multiple outbound parcels to resume, this will also block until the caretaker has handled all of them, as the `outboundParcels` channel is unbuffered.

I'm not sure if `Start()` being blocked on parcel resume would cause any issues right now though.

Reply:
Thanks for picking up on that. Good points. I think I'll put up a separate PR to run `resumePendingParcels` in a separate goroutine. I don't think this PR changes the sync problem you're outlining (I think it existed before this PR).

Reply:
https://github.com/lightninglabs/taproot-assets/pull/1068